### ISSUE TITLE
can pyroSAR work with tiff files or ee.Image object directly ?

### ISSUE THREAD

**Issue by ashishgitbisht** [johntruckenbrodt/pyroSAR#190]
- which operating system are you using?  
e.g. Windows 10, Ubuntu 18.4, etc.
- which environment is pyroSAR running in?  
e.g. system-wide Python installation, Anaconda environment, virtual environment, etc.
- which version of pyroSAR are you using?  
one installed via conda, pip or a clone of the GitHub repository?
-  which function of pyroSAR did you call with which parameters?  
- if applicable, which version of SNAP or GAMMA are you using in pyroSAR?
- the full error message

**Comment by johntruckenbrodt** [comment]
Hi @ashishgitbisht please provide more information than a question in the headline and the `bug` label.

**Comment by ashishgitbisht** [comment]
hello!!
*Query : I am working with imageCollection from Google earth engine, i am
not downloading the datasets of sentinel 1.*
*I am directly calling it using geemap API for earth engine.*

*I wanted to use pyroSAR, for pre-processing the dataset. so when i used
'identify' function from pyroSAR, it says object should be a string.*


*Thank you!!!!*

**Comment by pbrotoisworo** [comment]
I'm curious, what are you trying to do with the data? Why can't you process it in Earth Engine? You need a specialized Earth Engine API like [geemap](https://geemap.org/) or the official [EE package](https://developers.google.com/earth-engine/guides/python_install) to interact with Earth Engine data. It is designed for cloud computing.

[Earth Engine already preprocesses Sentinel-1 data](https://developers.google.com/earth-engine/guides/sentinel1#sentinel-1-preprocessing).

**Comment by ashishgitbisht** [comment]
> 

Yes I am already using geemap to integrate Earth engine with Python in jupyter notebook.
Also the earth engine data is good and pre processed but it doesn not qualify to be ARD (analysis ready data) as per CARD4L.
I applied certain pre-processing on S1-GRD to convert it into ARD data, that is done in earth engine, but then i came across pyroSAR. So i was interested to be able to use pyroSAR for pre-processing dataset.
Interesting fact is I dont have dataset stored in local directory, its rather a python object or image collection object which has dataset.
How do we then pass into pyroSAR for further pre-processing to make it ARD data???
thats largely my query.
thanks for prompt responses!!

**Comment by pbrotoisworo** [comment]
You cannot use pyroSAR with Earth Engine directly. One of the limitations of Earth Engine is that if you are using their cloud resources it only allows the methods in the official documentation. It will not allow external software to modify the data on their server.

https://developers.google.com/earth-engine/apidocs/ee-image
![image](https://user-images.githubusercontent.com/32087633/152095049-9933a7ea-eca3-4bd8-a1de-22e612508dd8.png)

**Comment by johntruckenbrodt** [comment]
Hi @ashishgitbisht and @pbrotoisworo.
I am not so familiar with GEE to be honest. The last thing I know is that the data is processed but without the RTC correction ([Small 2011](https://ieeexplore.ieee.org/document/5752845)).
RTC correction has to be done in radar geometry so it is not possible with the already geocoded GEE data. There are however other methods to apply terrain correction to GEE-processed images, e.g. [Vollrath et al. 2020](https://www.mdpi.com/2072-4292/12/11/1867). Andreas Vollrath is the main developer of [OpenSarToolkit](https://github.com/ESA-PhiLab/OpenSarToolkit). Perhaps there you can find an implementation.
I don't see how working with GEE image objects in pyroSAR would work. As @pbrotoisworo pointed out, this lets you apply GEE operations on the platform, whereas pyroSAR needs access to a local file system.
It is a very interesting thought to extend pyroSAR to some GEE functionality. But this would require some amount of work to blend it in with the existing functionality. pyroSAR is currently not designed for such cloud workflows.
@ashishgitbisht I think you might be better off developing your own prototype for now. Once you have a working prototype we could talk again and see how we can merge it.
